docs/forwarding_logs_to_gateway.md

```diff
@@ -0,0 +1,119 @@
+# Forwarding Logs to Loki Operator
+
+This document will describe how to send application, infrastructure, and audit logs to the Lokistack Gateway as different tenants using Promtail. The gateway provides secure access to the distributor (and query-frontend) via consulting an OAuth/OIDC endpoint for the request subject.
+
+__Please read the [hacking_loki_operator.md](./hacking_loki_operator.md) document before proceeding with the following instructions.__
+
+_Note: While this document will only give instructions for two methods of log forwarding into the gateway, the examples given in the Promtail section can be extrapolated to other log forwarders._
+
+## Openshift Logging
+
+Although there is a way to [forward logs to an external Loki instance](https://docs.openshift.com/container-platform/4.9/logging/cluster-logging-external.html#cluster-logging-collector-log-forward-loki_cluster-logging-external), [Openshift Logging](https://github.com/openshift/cluster-logging-operator) does not currently have support to send logs through the Lokistack Gateway.
+
+Support will be added in the near future.
+
+## Forwarding Clients
+
+In order to enable communication between the client(s) and the gateway, follow these steps:
+
+1. Deploy the Loki Operator and an `lokistack` instance with the gateway enabled.
+
+2. Create a `ServiceAccount` to generate the `Secret` which will be used to authorize the forwarder.
+
+```console
+kubectl -n openshift-logging create serviceaccount <SERVICE_ACCOUNT_NAME>
+```
+
+3. Configure the forwarder and deploy it to the `openshift-logging` namespace.
+
+4. Create the following `ClusterRole` and `ClusterRoleBinding` which will allow the cluster to authenticate the user(s) submitting the logs:
+
+```yaml
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: lokistack-dev-tenant-logs-role
+rules:
+- apiGroups:
+  - 'loki.openshift.io'
+  resources:
+  - application
+  - infrastructure
+  - audit
+  resourceNames:
+  - logs
+  verbs:
+  - 'get'
+  - 'create'
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: lokistack-dev-tenant-logs-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: lokistack-dev-tenant-logs-role
+subjects:
+- kind: User
+  name: "kube:admin"
```

Comment:
I suggest to drop the `kube:admin` user here. It is a temporary user account that is usually dropped after configuring IDP on k8s.